File: editorial/chap08/answer_B54.cpp
#include <iostream>
#include <map>
using namespace std;
 
int N, A[100009];
map<int, int> Map;
 
int main() {
	// 入力
	cin >> N;
	for (int i = 1; i <= N; i++) cin >> A[i];
 
	// 答えを求める
	long long Answer = 0;
	for (int i = 1; i <= N; i++) {
		Answer += Map[A[i]];
		Map[A[i]] += 1;
	}
 
	// 出力
	cout << Answer << endl;
	return 0;
}

File: editorial/chap08/answer_B54.py
Map = {}
 
# 入力
N = int(input())
A = [int(input()) for _ in range(N)]

# 答えを求める
Answer = 0
for a in A:
	# dict.get にはキーが存在しなかった場合のデフォルト値を設定できる
	Answer += Map.get(a, 0)
	Map[a] = Map.get(a, 0) + 1

# 出力
print(Answer)
